### ISSUE TITLE
ArcFace: Additive Angular Margin Loss for Deep Face Recognition

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#112]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1801.07698
- 公開日時：2018/01/23
- 組織 : 
- 被引用数（記事作成時点）：1045 件
- 実装コード（<公式>）：https://github.com/deepinsight/insightface
- 実装コード（<非公式>PyTorch）：https://github.com/ronghuaiyang/arcface-pytorch
- Publication : 

## 1. どんなもの？
- クラス間分散が大きくクラス内分散を小さくなるように softmax loss を改良した損失関数を提案。DNN を用いた顔認識タスクにおいて、SOTA を実現
    <image src="https://user-images.githubusercontent.com/25688193/101271996-8d184980-37cb-11eb-9a0d-6dcf6fb37b6e.png" width="890">

## 2. 先行研究と比べてどこがすごいの？
- 従来の DNN を用いた顔認識モデルの損失関数としては、softmax loss が広く採用されていた。
	しかしながら、softmax loss で学習されたネットワークでは、各特徴量のクラス間分散が小さくなるように分布され、分類境界付近では曖昧な分類結果となる。このことは、顔認識モデルなどの所属クラスを識別するタスクでは問題となる。
	本手法では、クラス間分散が大きくクラス内分散を小さくなるように softmax loss を改良することで、DNN を用いた顔認識タスクにおいて、SOTA を実現している。

## 3. 技術や手法の"キモ"はどこにある？
- 従来の softmax loss の問題点
	従来の顔認識モデルの損失関数には、以下の式で定義される softmax loss が広く採用されてきた。
    <image src="https://user-images.githubusercontent.com/25688193/101272023-c650b980-37cb-11eb-8a4a-351315bda27a.png" width="565">

	この softmax loss を、簡単のため <image src="https://user-images.githubusercontent.com/25688193/101272031-dbc5e380-37cb-11eb-9dd5-cbe425386698.png" width="256"> として変形すると、
	以下の式のようになる。
    <image src="https://user-images.githubusercontent.com/25688193/101272038-ef714a00-37cb-11eb-94ab-83613efcc287.png" width="416">

	この softmax loss の変形式より、softmax loss で学習されたネットワークにおける中間特徴量は、左下図のような超球面を構成することがわかる。
    <image src="https://user-images.githubusercontent.com/25688193/101272048-06b03780-37cc-11eb-8a58-71cc6ce5aac7.png" width="512">

	この図より、softmax loss では各特徴量を大まかに分類するが、分類境界が曖昧になっていることがわかる。
	言い換えると、クラス間分散が小さくなっており、このことは顔認識モデルなどで各クラスを識別 or 判別する際には問題となる。
	そのため arcface では上図 (b) のように、クラス間分散が大きくクラス内分散を小さくなるように softmax loss の改良を行う。

- arcface の損失関数
	前述のように、従来の softmax loss では、クラス間分散が小さくなるために、顔認識モデルなどで各クラスを識別する際には不適切であるという問題が存在する。
	arcface では、クラス間分散が大きくクラス内分散を小さくなるように softmax loss の改良を行う。
    <image src="https://user-images.githubusercontent.com/25688193/101272057-2cd5d780-37cc-11eb-8b62-1cfb703be496.png" width="771">

	上図は、arcface の損失関数での処理を示した図である。
	具体的には、以下のような処理を行っている
	1. ネットワーク内の特徴量 x_i  に対して、L2正規化を行う。
	2. ネットワークの重みベクトル W_j∈W に対して、L2正規化を行う。
	3. 正解クラス y_i  の logit cos⁡(y_i )  のみに対して、マージン m を加えて、cos⁡(y_i )→cos⁡(y_i+m) とする。
		これにより、正解クラスに関してはの角度 θ がマージン m よりも大きくなるように特徴量 x が分布されるので
、クラス間分散が大きくなるように特徴量が分布（＝学習）される。
            <image src="https://user-images.githubusercontent.com/25688193/101272073-4d059680-37cc-11eb-8cf4-a09e7b1a77f9.png" width="325">

	これらの処理により定義される arcface の損失関数は、以下の式のようになる。
    <image src="https://user-images.githubusercontent.com/25688193/101272089-60186680-37cc-11eb-9934-bad1b6d8c977.png" width="508">
 
- ネットワーク構成
	arcface の損失関数は、一般的な DNN 全般で有効な損失関数になっているが、本論文では、ResNet-50, ResNet-101 のネットワーク構造を採用している。
	但し最終 conv 層の後に、BatchNorm + Dropout + FC + BathNorm 層を追加し、512 次元の特徴ベクトルを出力するようにしている。（x_i∈Rd  (d=512)）
	また、入力データの前処理として、５つの顔特徴で正規化した 112x112 の顔クロップ画像を生成する処理を行っている。

## 4. どうやって有効だと検証した？
- 学習用データセット
    <image src="https://user-images.githubusercontent.com/25688193/101272111-863e0680-37cc-11eb-9fa7-9c42db4ce00c.png" width="312">
- 既存の顔認識モデルとの定量比較検証
    <image src="https://user-images.githubusercontent.com/25688193/101272113-9c4bc700-37cc-11eb-8943-5c5275358417.png" width="350">
	上図は、LFW YTF データセットにおける検証用データでの正解率（％）を示した図である。
	ArcFace （ResNet-100）が最も優れたスコアを実現している。

## 5. 議論はあるか？
- arcface の損失関数で学習された ResNet は、多くの顔生成モデルの事前学習済み特徴量 encoder として採用されている。（例：論文「Encoding in Style: a StyleGAN Encoder for Image-to-Image Translation」など）

## 6. 次に読むべき論文はあるか？
- xxx

## 7. 参考文献
- https://qiita.com/yu4u/items/078054dfb5592cbb80cc
- https://speakerdeck.com/inoichan/arcfacefalseshi-zu-mi
